src/docs/customize/component-customization.mdx

```diff
@@ -0,0 +1,184 @@
+---
+name: Component Customization
+menu: Customize
+route: /customize/component-customization
+---
+
+# Component Customization
+
+## Global props
+
+import { Playground } from 'docz'
+import {
+  Toolbar,
+  ToolbarItem,
+} from '../../lib/components/layout/Toolbar';
+import Button from '../../lib/components/ui/Button';
+import SelectField from '../../lib/components/ui/SelectField';
+import TextField from '../../lib/components/ui/TextField';
+import { RUIProvider } from '../../lib/provider';
+
+React UI often provides more variants of components, so a user can
+pick the most suitable variant.
+
+There two ways to configure component:
+
+1. Pass props directly to a specific component.
+2. Wrap application (or its part) with `RUIProvider` and pass arbitrary props for
+a specific component through `globalProps` prop.
+
+### Basic Usage
+
+To define global props, you need to import `RUIProvider` first:
+
+```js
+import { RUIProvider } from '@react-ui-org/react-ui';
+```
+
+Then wrap application (or its part) with `RUIProvider` with `globalProps` prop object,
+where its keys are names of components and its values are objects with arbitrary
+props you want to pass to a specific component.
+
+Keys conform to actual names of components:
+
+```jsx
+<RUIProvider globalProps={{
+  Button: { priority: 'filled' },
+  TextField: { variant: 'filled' },
+  SelectField: { variant: 'filled' },
+}}>
+  //...
+</RUIProvider>
+```
+
+See working example:
+
+<Playground>
+  {() => {
+    const [variant, setVariant] = React.useState('filled');
+    return (
+      <RUIProvider
+        globalProps={{
+          Button: { priority: variant },
+          TextField: { variant },
+          SelectField: { variant },
+        }}
+      >
+        <Toolbar align="bottom">
+          <ToolbarItem>
+            <SelectField
+              changeHandler={(e) => setVariant(e.target.value)}
+              id="variant"
+              label="Select variant of Select Field"
+              options={[
+                {
+                  label: 'filled',
+                  value: 'filled',
+                },
+                {
+                  label: 'outline',
+                  value: 'outline',
+                },
+              ]}
+              value={variant}
+            />
+          </ToolbarItem>
+          <ToolbarItem>
+            <TextField
+              id="my-text-field"
+              label="Text Field"
+            />
+          </ToolbarItem>
+          <ToolbarItem>
+            <Button
+              clickHandler={() => {}}
+              id="my-button"
+              label="Button"
+            />
+          </ToolbarItem>
+        </Toolbar>
+      </RUIProvider>
+    );
+  }}
+</Playground>
+
+### Best practices
```

Comment:
I'm not sure about this section.

For example, when my entire project is just one `<Form>` I can imagine something  like:
```
<RUIProvider globalProps={{
  TextField: { clickHandler: setValue },
}}>
```
I would prefer to leave this up to the discretion of the developer using this feature.

Reply:
Maybe a different wording would help? Something like _however there are component props that are usually unique to their instance_…

Reply:
But why?

All props are handled exactly the same way. There is no technical reason for it. As far as non technical reasons that depends on the application. Sometimes it might be a terrible idea and sometimes it can be a simple and elegant solution.  Only the application author knows what props are to be unique and what props  can be shared in the given context.

`usually unique` always depends for whom... if all I ever do are single page three field forms over and over then my definition is very different from someone doing complex administrations.